src/tests/conformance/invalid_message/disconnect.rs

```diff
@@ -0,0 +1,303 @@
+    // zcashd: fail (message ignored)
+    // zebra:  fail (message ignored)
+    const PING_TIMEOUT: Duration = Duration::from_secs(1);
+
+    let mut node = Node::new().unwrap();
+    node.initial_action(Action::WaitForConnection)
+        .start()
+        .await
+        .unwrap();
+    // Create SyntheticNode which lets through Ping's
+    let mut synthetic_node = SyntheticNode::builder()
+        .with_full_handshake()
+        .with_message_filter(
+            MessageFilter::with_all_auto_reply().with_ping_filter(Filter::Disabled),
+        )
+        .build()
+        .await
+        .unwrap();
+
+    synthetic_node.connect(node.addr()).await.unwrap();
+
+    // Wait for a Ping request.
+    match synthetic_node.recv_message_timeout(PING_TIMEOUT).await {
+        Ok((_, Message::Ping(_))) => synthetic_node
+            .send_direct_message(node.addr(), Message::Pong(Nonce::default()))
+            .await
+            .unwrap(),
+        Ok((_, message)) => {
+            panic!(
+                "Unexpected message while waiting for Ping: {}",
+                message.short_string()
+            );
+        }
+        Err(err) => {
+            panic!("Error waiting for Ping: {:?}", err);
+        }
+    }
+
+    // Use Ping-Pong to check node's response.
+    // We expect a disconnect.
+    match synthetic_node
+        .ping_pong_timeout(node.addr(), DC_TIMEOUT)
+        .await
+    {
+        Err(PingPongError::ConnectionAborted) => {}
+        Ok(_) => panic!("Message was ignored."),
+        Err(err) => panic!("Connection was not aborted: {:?}", err),
+    }
+
+    synthetic_node.shut_down();
+    node.stop().unwrap();
+}
+
+#[tokio::test]
+async fn get_data_with_mixed_types() {
+    // zcashd: fail (replies with Block)
+    // zebra:  pass
+    let genesis_block = Block::testnet_genesis();
+    let mixed_inv = vec![genesis_block.inv_hash(), genesis_block.txs[0].inv_hash()];
+    let message = Message::GetData(Inv::new(mixed_inv));
+    run_test_case_message(message).await.unwrap();
+}
+
+#[tokio::test]
+async fn inv_with_mixed_types() {
+    // zcashd: fail (message ignored)
+    // zebra:  pass
+
+    // Inv with mixed inventory (using non-genesis block since all node's "should" have genesis already,
+    // which makes advertising it non-sensical).
+    let block_1 = Block::testnet_1();
+    let mixed_inv = vec![block_1.inv_hash(), block_1.txs[0].inv_hash()];
+    let message = Message::Inv(Inv::new(mixed_inv));
+    run_test_case_message(message).await.unwrap();
+}
+
+#[tokio::test]
+async fn addr_without_timestamp() {
+    // zcashd: pass
+    // zebra:  fail (replies with Reject(Malformed))
+
+    // Encode a custom type which mimics Message::Addr but without the timestamp.
+    let bytes = AddrWithoutTimestamp::new().encode().unwrap();
+
+    run_test_case_bytes(bytes).await.unwrap();
+}
+
+/// Mimics the encoding of [`NetworkAddr`] but excludes the timestamp field.
+struct NetAddrWithoutTimestamp(NetworkAddr);
```

Comment:
For consistency with `NetworkAddr`:
```suggestion
struct NetworkAddrWithoutTimestamp(NetworkAddr);
```